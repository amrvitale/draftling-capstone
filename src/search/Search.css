* {
    box-sizing: border-box;
  }
  
  #draftlingSearchResultContainer {
    display: flex;
    flex-direction: row;
    flex-flow: row wrap;
    align-items: center;
    justify-content: space-evenly;
}

.single-draftling {
    margin: 1px;
    padding: 5px;
    border-radius: 5px;
    background-color: white;
    justify-content: space-evenly;
}

#draftlingSearchResultContainer .single-draftling {
    padding: 3px;
    display: flex;
}

div.single-draftling {
    margin: 30px;
    justify-self: center;
    height: 150px;
    width: 320px;
    box-shadow: 3px 3px 5px 6px #ccc;
}
.searchPageHero {
    background-image: url(./search.jpg);
    background-position: center; /* Center the image */
    background-repeat: no-repeat; /* Do not repeat the image */
    background-size: cover; /* Resize the background image to cover the entire container */
}

.searchHeading {
    -webkit-text-fill-color: rgb(37, 26, 26);    
    font-family: 'Playfair Display', 'Georgia', serif;
}

.titleSearch, .wordcountSearch, .genreSearch {
    font-size: 30px;
    color: whitesmoke;
}

input, select {
    width: 100px;
}
