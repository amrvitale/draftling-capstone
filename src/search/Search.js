import React from 'react';
import './Search.css';
import { Link } from 'react-router-dom';
import Published from '../published/Published'
import config from '../config'

class Search extends React.Component {
    state = {
        searchValue: '',
        wordcountValue: '',
        genreValue: '',
        draftlings: []
    };

handleOnChange = event => {
    this.setState({ searchValue: event.target.value });
    this.setState( {wordcountValue: event.target.value});
    this.setState( {genreValue: event.target.value});
};

handleSearch = () => {
    this.makeAPICall(this.state.searchValue);
    this.makeAPICall(this.state.wordcountValue);
    this.makeAPICall(this.state.genreValue);
}

makeAPICall = searchInput => {
    fetch(`${config.API_ENDPOINT}/search`, {
        method: 'GET',
        headers: {
            'content-type': 'application/json'
        },   
    })
    .then(res => {
        return res.json();
    })

    .then(draftling => {
        this.setState( { draftlings: draftling.draftlings})
    })

    .catch(error => {
        console.log({ error })
    })
}

    render() {
    
        return (
            <div className="searchPage">
                <h1>Search Draftlings</h1>
                <br />
                <br />
                <section className="titleSearch">
                    <label htmlFor="titleSearch">Title:</label>
                    <input 
                        type="text" 
                        placeholder="Search by title" 
                        onChange={event => this.handleOnChange(event)} 
                        value={this.state.searchValue}
                     />
                </section>
                <br />
                <section className="wordcountSearch">
                    <label htmlFor="wordcountSearch">Word Count:</label>
                    <select 
                        name="wordcount" 
                        id="wordcount" 
                        onChange={event => this.handleOnChange(event)} 
                        value={this.state.wordcountValue}
                     >
                        <option></option>
                        <option value="Six word story">Six word story</option>
                        <option value="Minisaga">Minisaga: 50 words</option>
                        <option value="Micro fiction">Microfiction: 51 - 100 words</option>
                        <option value="Twitterature">Twitterature: 280 characters</option>
                        <option value="Sudden Fiction">Sudden Fiction: 300 - 750 words</option>
                        <option value="Flash Fiction">Flash Fiction: 751 - 1000 words</option>
                        <option value="Short Story">Short Story: 1001 - 7500 words</option>
                        <option value="Novellete">Novellete: 7501 - 20,000 words</option>
                        <option value="Novella">Novella: 20,001 - 60,000 words</option>
                        <option value="Novel">Novel: 60,0001 + words</option>
                    </select>
                </section>
                <br />
                <section className="genreSearch">
                    <label htmlFor="genreDrop">Search by Fiction Genre</label>
                    <select 
                        name="genreDrop" 
                        id="genreDrop"
                        onChange={event => this.handleOnChange(event)} 
                        value={this.state.genreValue}
                    >
                        <option></option>
                        <option value="action / adventure">Action / Adventure</option>
                        <option value="classic">Classic</option>
                        <option value="contemporary">Contemporary</option>
                        <option value="crime/detective">Crime / Detective</option>
                        <option value="fable">Fable</option>
                        <option value="fairy tale">Fairy Tale</option>
                        <option value="fantasy">Fantasy</option>
                        <option value="folktale">Folktale</option>
                        <option value="gothic">Gothic</option>
                        <option value="historical fiction">Historical Fiction</option>
                        <option value="horror">Horror</option>
                        <option value="humor">Humor</option>
                        <option value="magical realism">Magical realism</option>
                        <option value="mystery">Mystery</option>
                        <option value="romance">Romance</option>
                        <option value="science fiction">Science Fiction</option>
                        <option value="thriller">Thriller</option>
                        <option value="western">Western</option>
                    </select>
                </section>

                    <button type="submit" onClick={this.handleSearch}>Search</button>
                    {this.state.draftlings ? (
                        <div id="draftlingSearchResultContainer">
                            {this.state.draftlings.map((draftling, index) => (
                                <div class="single-draftling" key={index}>
                                </div>
                            ))}
                        </div>
                    ) : (
                        <p>Search for a draftling</p>
                    )}
            </div>
        );
    }
}

export default Search;


