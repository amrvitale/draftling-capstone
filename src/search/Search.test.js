import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Search from './Search'

describe (`Search component`, () => {
    it('renders a .Search by default', () => {
        const wrapper = shallow(<Search />)
        expect (toJson(wrapper)).toMatchSnapshot()
    })

    
})