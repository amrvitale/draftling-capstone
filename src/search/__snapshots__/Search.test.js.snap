// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders a .Search by default 1`] = `
<div
  className="searchPage"
>
  <div
    className="searchPageHero"
  >
    <h1
      className="searchHeading"
    >
      Search Draftlings
    </h1>
  </div>
  <br />
  <br />
  <section
    className="titleSearch"
  >
    <label
      htmlFor="titleSearch"
    >
      Title:
    </label>
    <input
      id="titleSearch"
      name="titleSearch"
      onChange={[Function]}
      type="text"
      value=""
    />
  </section>
  <br />
  <section
    className="wordcountSearch"
  >
    <label
      htmlFor="wordcountSearch"
    >
      Word Count:
    </label>
    <select
      id="wordcountSearch"
      name="wordcountSearch"
      onChange={[Function]}
      value=""
    >
      <option />
      <option
        value="Six word story"
      >
        Six word story
      </option>
      <option
        value="Minisaga"
      >
        Minisaga: 50 words
      </option>
      <option
        value="Micro fiction"
      >
        Microfiction: 51 - 100 words
      </option>
      <option
        value="Twitterature"
      >
        Twitterature: 280 characters
      </option>
      <option
        value="Sudden Fiction"
      >
        Sudden Fiction: 300 - 750 words
      </option>
      <option
        value="Flash Fiction"
      >
        Flash Fiction: 751 - 1000 words
      </option>
      <option
        value="Short Story"
      >
        Short Story: 1001 - 7500 words
      </option>
      <option
        value="Novellete"
      >
        Novellete: 7501 - 20,000 words
      </option>
      <option
        value="Novella"
      >
        Novella: 20,001 - 60,000 words
      </option>
      <option
        value="Novel"
      >
        Novel: 60,0001 + words
      </option>
    </select>
  </section>
  <br />
  <section
    className="genreSearch"
  >
    <label
      htmlFor="genreDrop"
    >
      Fiction Genre
    </label>
    <select
      id="genreDrop"
      name="genreDrop"
      onChange={[Function]}
      value=""
    >
      <option />
      <option
        value="Action/Adventure"
      >
        Action/Adventure
      </option>
      <option
        value="Classic"
      >
        Classic
      </option>
      <option
        value="Contempory"
      >
        Contempory
      </option>
      <option
        value="Crime/Detective"
      >
        Crime/Detective
      </option>
      <option
        value="Fable"
      >
        Fable
      </option>
      <option
        value="Fairy Tale"
      >
        Fairy Tale
      </option>
      <option
        value="Folktale"
      >
        Folktale
      </option>
      <option
        value="Gothic"
      >
        Gothic
      </option>
      <option
        value="Historical Fiction"
      >
        Historical Fiction
      </option>
      <option
        value="Horror"
      >
        Horror
      </option>
      <option
        value="Humor"
      >
        Humor
      </option>
      <option
        value="Magical Realism"
      >
        Magical realism
      </option>
      <option
        value="Mystery"
      >
        Mystery
      </option>
      <option
        value="Romance"
      >
        Romance
      </option>
      <option
        value="Science Fiction"
      >
        Science Fiction
      </option>
      <option
        value="Thriller"
      >
        Thriller
      </option>
      <option
        value="Western"
      >
        Western
      </option>
    </select>
  </section>
  <button
    onClick={[Function]}
    type="submit"
  >
    Search
  </button>
  <br />
  <br />
  <br />
  <br />
</div>
`;
